src/vivarium/framework/randomness/stream.py

```diff
@@ -254,18 +260,28 @@ def filter_for_probability(
         if population.empty:
             return population
 
-        index = population if isinstance(population, pd.Index) else population.index
+        if isinstance(population, pd.Index):
+            index = population
+        else:
+            index = population.index
```

Comment:
Why was this change necessary?

Reply:
mypy gave "pd.Index has no attribute index" even though you think it would be smart enough.